tests/unit/test-class-wc-payments-webhook-processing-service.php

```diff
@@ -1733,4 +1733,31 @@ public function provider_mode_mismatch_detection() {
 			'No mode proceeds'                => [ null, $this->once() ],
 		];
 	}
+
+	public function test_process_throws_exception_when_order_not_found_for_intent_id() {
```

Comment:
Could you also replicate a test and add a case where `WC_Order_Refund` is returned so we cover all possible returns?

Reply:
Added a test here [cfb7801](https://github.com/Automattic/woocommerce-payments/pull/9761/commits/cfb7801f4e5068f3d2559db00d37ce3b1bc7dd27) but please check my previous comment.